sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -501,4 +513,88 @@ trait Row extends Serializable {
   private def getAnyValAs[T <: AnyVal](i: Int): T =
     if (isNullAt(i)) throw new NullPointerException(s"Value at index $i is null")
     else getAs[T](i)
+
+  /** The compact JSON representation of this row. */
+  def json: String = compact(jsonValue)
+
+  /** The pretty (i.e. indented) JSON representation of this row. */
+  def prettyJson: String = pretty(render(jsonValue))
+
+  /**
+   * JSON representation of the row.
+   *
+   * Note that this only supports the data types that are also supported by
+   * [[org.apache.spark.sql.catalyst.encoders.RowEncoder]].
+   *
+   * @return the JSON representation of the row.
+   */
+  private[sql] def jsonValue: JValue = {
+    require(schema != null, "JSON serialization requires a non-null schema.")
+
+    lazy val timeZone = TimeZone.getTimeZone(SQLConf.get.sessionLocalTimeZone)
+    lazy val dateFormatter = DateFormatter.apply(timeZone.toZoneId)
```

Comment:
Why do you use explictit `.apply`?

Reply:
Why not? In some cases it is clearer. However the only reason I am responding to this is because IMO we shouldn't dwell too much on style nits, or wait with them until a final review round.

Reply:
> Why not? In some cases it is clearer.

Thanks for the answer. I was just wondered why you desugared this code, but didn't desugar `match ... case`, for instance. ;-)